packages/phone-number-privacy/combiner/src/pnp/endpoints/quota/action.ts

```diff
@@ -50,35 +70,36 @@ export function createPnpQuotaHandler(
       responseSchema: PnpQuotaResponseSchema,
       shouldCheckKeyVersion: false,
     })
-    const warnings = logPnpSignerResponseDiscrepancies(logger, signerResponses)
+    warnings.push(...logPnpSignerResponseDiscrepancies(logger, signerResponses))
 
     const { threshold } = keyVersionInfo
 
     if (signerResponses.length >= threshold) {
       try {
         const quotaStatus = findCombinerQuotaState(keyVersionInfo, signerResponses, warnings)
-        send(
-          response,
-          {
+
+        if (quotaStatus.performedQueryCount === quotaStatus.totalQuota) {
+          // Sets that there is not more quota for that account
+          noQuotaCache.setNoMoreQuota(account, quotaStatus.totalQuota)
```

Comment:
If we stored both performedQueryCount and totalQuota we could return responses on the quota endpoint even when quota hasn't been exhausted. Maybe not worth doing now but could add it to a TODO

Reply:
The problem with that is that you are assuming only one combiner.
As we could eventually have more than one source (it will happen to us with the v1 <> v2), you won't actually know the quota of the signer.
The only one available to track the quota is the signer. That's why the only thing that we can know, is that if the signer has no more quota, that will be probably the same scenario until a new block is mined, which will be in 5 secs.

Reply:
makes sense!